product_customer_code_picking/models/stock_picking.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013-2017 Agile Business Group sagl
+#     (<http://www.agilebg.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    @api.multi
+    def _get_product_customer_code(self):
+        product_supplierinfo_obj = self.env['product.supplierinfo']
+        for move in self:
```

Comment:
Clearer to use something like :

`for move in self.filtered(
lambda m: m.picking_id and m.picking_id.partner_id and m.product_id.supplier_ids):`
    `supplier_info = product_supplierinfo_obj.search('product_tmpl_id', '=', move.product_id.product_tmpl_id.id)`
(...)